src/lightning/fabric/fabric.py

```diff
@@ -50,6 +50,10 @@
 from lightning.fabric.wrappers import _FabricDataLoader, _FabricModule, _FabricOptimizer, _unwrap_objects
 
 
+def _do_nothing(*_: Any) -> None:
+    pass
```

Comment:
I don't like that this is at the top, seems odd. Any workarounds?

Reply:
Importing it from somewhere else

Reply:
But previously we used None as the default and that worked. We can do

`function = _do_nothing if function is None else function`

Reply:
Your logic will still work and as a bonus you can still use the patching in the test.

Reply:
Your suggestion won't work because then we cannot differentiate `launch()` from `launch(None)` which is one of the main motivations of this PR.

Let me know if you want me to move `_do_nothing` somewhere else and import it

Reply:
To make your input validation, you don't need to differentiate between `launch() from launch(None)`. This could work:

```python
def launch(function):
    function = _do_nothing if function is None else function
    
    if strategy is xla:
         if function is do_nothing or not callable(function):
             # error, you need to pass in a function
    else:
        ...
```